contribution-guide.md

```diff
@@ -0,0 +1,143 @@
+
+1. In Eclipse, click **File** and then click **Import..**.
+2. Under the **Maven** option, select **Existing Maven Projects**. 
+3. Browse the file system and select the Maven module to open it.
+
+## Contributing to Ballerina grammar
+
+Ballerina grammar has been implemented using ANTLR4. To get a basic understanding on ANTLR 4 grammar syntax and concepts before working with Ballerina grammar, go to [Parr, Terence (January 15, 2013), The Definitive ANTLR 4 Reference](#http://lms.ui.ac.ir/public/group/90/59/01/15738_ce57.pdf).
+
+[Ballerina grammar](#https://github.com/ballerina-platform/ballerina-lang/tree/master/compiler/ballerina-lang/src/main/resources/grammar) consists of two files:
+
+- **BallerinaLexer.g4:** contains the lexer rules for Ballerina grammar. Lexer is responsible for tokenizing an input Ballerina source code.
+- **BallerinaParser.g4:** contains the parser rules. Parser listens to the token stream generated by the lexer. High level grammar productions/abstractions are defined in the parser using those tokens.
+
+Once a change is done to any of the grammar files, the lexer and the parser need to be re-generated. To generate the lexer and the parser, navigate to the <BALLERINA_LANG_REPO>/compiler/ballerina-lang/src/main/resources/grammar/ directory, and execute the below command. 
+
+>**Tip:** Download the [antlr-complete-4.5.3.jar](#https://jar-download.com/artifacts/org.antlr/antlr4/4.5.3/source-code) file and replace <PATH-TO-ANTLR-JAR> in the below command with the location in which you saved it.
+
+```bash 
+java -jar <PATH-TO-ANTLR-JAR>/antlr-4.5.3-complete.jar *.g4 -package org.wso2.ballerinalang.compiler.par
+```
+
+For more information about the Ballerina compiler, go to [Ballerina Compiler — Design](#https://medium.com/@sameerajayasoma/ballerina-compiler-design-3406acc2476c?).
+
+## Submitting your contribution
+
+1. Do your changes in the <BALLERINA_LANG_REPO> using Eclipse or IntelliJ IDEA.
+2. Add unit tests accordingly. (The build process will automatically execute all the tests.)
+3. Commit the changes to your Fork and push them to the original <a href="https://github.com/ballerina-platform/ballerina-lang/issues">ballerina-lang</a> repo by sending a Pull Request. 
+
+>**Tip:** Follow these commit message requirements:
+
+- Separate subject from body with a blank line
+- Limit the subject line to 50 characters
+- Capitalize the subject line
+- Do not end the subject line with a period
+- Use the imperative mood in the subject line
+- Wrap the body at 72 characters
+- Use the body to explain what and why vs. how
+
+For more information, go to [How to Write a Git Commit Message](#https://chris.beams.io/posts/git-commit/).
+
+## Accepting the Contributor License Agreement 
+
+You need to Accept the Ballerina Contributor License Agreement (CLA) here when prompted by GitHub while sending your first Pull Request (PR. Subsequent PRs will not require CLA acceptance.
```

Comment:
```suggestion
You need to Accept the Ballerina Contributor License Agreement (CLA) here when prompted by GitHub while sending your first Pull Request (PR). Subsequent PRs will not require CLA acceptance.
```

Reply:
Fixed as follows:

"You need to Accept the Ballerina Contributor License Agreement (CLA) here when prompted by GitHub while sending your first Pull Request."